ipatests/test_integration/test_otp.py

```diff
@@ -392,6 +455,103 @@ def test_2fa_disable_single_prompt(self):
             self.master.run_command(['semanage', 'login', '-D'])
             sssd_conf_backup.restore()
 
+    def test_2fa_only_with_password(self):
+        """Test ssh with 2FA only with the password(first factor) when
+        user-auth-type is opt and password.
+
+        Test for :  https://github.com/SSSD/sssd/pull/7500
+
+        Add the IPA user and user-auth-type set to opt and password.
+        Authenticate the user only with password, just press enter
+        at `Second factor`
+        """
+
+        master = self.master
+        USER3 = 'sshuser3'
+        sssd_conf_backup = tasks.FileBackup(master, paths.SSSD_CONF)
+        first_prompt = 'Enter first factor:'
+        second_prompt = 'Enter second factor:'
+        add_contents = textwrap.dedent('''
+            [prompting/2fa/sshd]
+            single_prompt = False
+            first_prompt = {0}
+            second_prompt = {1}
+            ''').format(first_prompt, second_prompt)
+        set_sssd_conf(master, add_contents)
+        tasks.create_active_user(master, USER3, PASSWORD)
+        tasks.kinit_admin(master)
+        tasks.clear_sssd_cache(self.master)
+        master.run_command(['ipa', 'user-mod', USER3, '--user-auth-type=otp',
+                            '--user-auth-type=password'])
+        try:
+            otpuid, totp = add_otptoken(master, USER3, otptype='totp')
+            master.run_command(['ipa', 'otptoken-show', otpuid])
+            totp.generate(int(time.time())).decode('ascii')
+            otpvalue = "\n"
+            tasks.clear_sssd_cache(self.master)
+            result = ssh_2fa_with_cmd(master, self.master.external_hostname,
+                                      USER3, PASSWORD, otpvalue=otpvalue,
+                                      command="klist")
+            print(result.stdout_text)
+            assert 'Password authentication successful' in result.stdout_text
+            assert USER3 in result.stdout_text
+            self.master.domain.realm
```

Comment:
This line triggers a lint error:
************* Module ipatests.test_integration.test_otp
ipatests/test_integration/test_otp.py:498: [W0104(pointless-statement), TestOTPToken.test_2fa_only_with_password] Statement seems to have no effect)

Reply:
updated.